lib/query/lib/prepareForProcess.js

```diff
@@ -44,12 +44,40 @@ function applyPagination(body, _params) {
     }
 }
 
-export default (_body, _params = {}) => {
+function handleTextFilters(data, isClient) {
+    _.each(data, field => {
+        if (field.$text) {
+            if (isClient) {
+                let $or = _.map(field.$text.$fields, (val, key) => {
+                    return {
+                        [key]: {
+                            $regex:field.$text.$search,
+                            $options: field.$text.$caseSensitive ? '' : 'i'
+                        }
+                    }
+                })
+                delete field.$text
+                if (_.isEmpty(field)) {
+                    field = {$or}
```

Comment:
doing this will not change data at all. you just set the field.

Also there is no such thing as { field: { $or: [] } }

This simply cannot work. And also how do we treat the situation where we already have $or filters or $and filters :)